--- a/drivers/pmdomain/qcom/rpmhpd.c
+++ b/drivers/pmdomain/qcom/rpmhpd.c
@@ -31,6 +31,8 @@
  * @peer:		A peer power domain in case Active only Voting is
  *			supported
  * @active_only:	True if it represents an Active only peer
+ * @all_state_votes:	Send ACTIVE, WAKE and SLEEP votes even without an
+ *			active-only peer
  * @corner:		current corner
  * @active_corner:	current active corner
  * @enable_corner:	lowest non-zero corner
@@ -50,5 +52,6 @@ struct rpmhpd {
 	struct rpmhpd	*peer;
 	const bool	active_only;
+	const bool	all_state_votes;
 	unsigned int	corner;
 	unsigned int	active_corner;
 	unsigned int	enable_corner;
@@ -137,6 +140,18 @@ static struct rpmhpd lmx = {
 	.res_name = "lmx.lvl",
 };
 
+static struct rpmhpd sm8250_lcx = {
+	.pd = { .name = "lcx", },
+	.res_name = "lcx.lvl",
+	.all_state_votes = true,
+};
+
+static struct rpmhpd sm8250_lmx = {
+	.pd = { .name = "lmx", },
+	.res_name = "lmx.lvl",
+	.all_state_votes = true,
+};
+
 static struct rpmhpd mmcx_ao;
 static struct rpmhpd mmcx = {
 	.pd = { .name = "mmcx", },
@@ -535,7 +550,7 @@ static struct rpmhpd *sm8250_rpmhpds[] = {
 	[RPMHPD_EBI] = &ebi,
 	[RPMHPD_GFX] = &gfx,
-	[RPMHPD_LCX] = &lcx,
-	[RPMHPD_LMX] = &lmx,
+	[RPMHPD_LCX] = &sm8250_lcx,
+	[RPMHPD_LMX] = &sm8250_lmx,
 	[RPMHPD_MMCX] = &mmcx,
 	[RPMHPD_MMCX_AO] = &mmcx_ao,
 	[RPMHPD_MX] = &mx,
@@ -996,9 +1011,10 @@ static int rpmhpd_aggregate_corner(struct rpmhpd *pd, unsigned int corner)
 
 	pd->active_corner = active_corner;
 
-	if (peer) {
+	if (peer)
 		peer->active_corner = active_corner;
 
+	if (peer || pd->all_state_votes) {
 		ret = rpmhpd_send_corner(pd, RPMH_WAKE_ONLY_STATE,
 					 active_corner, false);
 		if (ret)
